subprojects/execution/src/main/java/org/gradle/internal/execution/ExecutionEngine.java

```diff
@@ -97,6 +97,10 @@ interface Execution {
         // TODO Parametrize UnitOfWork with this generated result
         @Nullable
         Object getOutput();
+
+        default boolean canStoreOutputsInCache() {
+            return true;
+        }
```

Comment:
:x: Add documentation here as well.

Reply:
Done in 6f1d0728d6f4ea6ab55a751949e6298756aa51b9.